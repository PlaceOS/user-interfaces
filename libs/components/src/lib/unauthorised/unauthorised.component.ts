import { Component } from '@angular/core';

@Component({
  selector: 'app-unauthorised',
  template: `
    <div class="unauthorised">
        <h1>403</h1>
        <h3 i18n="@@forbiddenLabel">Access forbidden.</h3>
        <p i18n="@@invalidPermissionLabel">You do not have permission to view this page and your access attempt has been recorded.</p>
        <p i18n="@@contactAdminLabel">Contact your Administrator if you feel that you should have access.</p>
    </div>
    `,
    styles: [`
        :host {
            display: flex;
            align-items: center;
            justify-content: center;
        }

        .unauthorised {
            width: 24em;
            background-color: #fff;
            color: rgba(#000, .85);
            border-radius: 4px;
            border: 1px solid #ccc;
            text-align: center;
            padding: .5em 2em 2em;
            margin: 1em;
        }

        @media (prefers-color-scheme: dark) {
            .unauthorised {
                background-color: #262626;
                color: #fff;
            }
        }
    `]
})
export class UnauthorisedComponent { }
