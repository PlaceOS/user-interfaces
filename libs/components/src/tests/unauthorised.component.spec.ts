import { Spectator, createComponentFactory } from '@ngneat/spectator/jest';

import { UnauthorisedComponent } from '../lib/unauthorised.component';

describe('UnauthorisedComponent', () => {
    let spectator: Spectator<UnauthorisedComponent>;
    const createComponent = createComponentFactory(UnauthorisedComponent);

    beforeEach(() => (spectator = createComponent()));

    it('should create component', () => {
        expect(spectator.component).toBeTruthy();
    });

    it('should match snapshot', () => {
        expect(spectator.element).toMatchSnapshot();
    });
});
