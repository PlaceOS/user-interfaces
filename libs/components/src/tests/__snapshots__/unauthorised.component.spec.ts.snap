// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnauthorisedComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="unauthorised rounded m-4 text-center bg-white text-black border border-gray-400 p-4 shadow"
  >
    <h1
      class="text-4xl"
    >
      403
    </h1>
    <h3
      class="text-2xl"
    >
      Access forbidden.
    </h3>
    <p
      class="py-4"
    >
       You do not have permission to view this page and your access attempt has been recorded. 
    </p>
    <p>
       Contact your Administrator if you feel that you should have access. 
    </p>
  </div>
</div>
`;
